### DISCUSSION TITLE
How to get value of specific attribute in nested collections?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#844] [Q&A] [answered]
Hi

While testing nested collections, based on cats demo: [codepen](https://codepen.io/galinhalx/pen/poaJJRa?editors=1100)

How can I get a specific value by its position in the JSON tree?

something like (doesn’t work):
[friends[0].name]

Are there other recommended demos, with nested collections to look at?

**Comment by GalinhaLX** [comment]
I saw there's
[first(1,friends.name)]    to get first item's value, same for [last](https://mavo.io/docs/functions/#last)

but can't find how to get 2nd value in a list of 10 ...

**Reply by DmitrySharabin** [reply]
> but can't find how to get 2nd value in a list of 10

Something like this should work: `[get(friends.name, 1)]`. Why does it work? [The get() function](https://mavo.io/docs/functions/#get) supports numeric values as the second parameter when working with collections (lists). In that case, the parameter is the _index_ of a collection item. Collection items are numbered starting from `0`. So, the second item has an index of `1`.

Hope it helps!

**Comment by GalinhaLX** [comment]
Perfect! it works , thanks @DmitrySharabin  

...by the way, it seems that all queries (when result is inside a nested collection) don't want to work:

see updated codepen [here](https://codepen.io/galinhalx/pen/poaJJRa?editors=1000), lines in red with prefix NESTED


In this case:
```
"cat": [{
        "name": "Vector",
        "age": 8,
        "toys": [1, 5, 7],
        "friends": [{
          "name": "Poo",
          "color": "blue"
        }, {
          "name": "bee",
          "color": "yellow"
        }]
```


How to solve those cases, only where 'something else' in tree, like:


Total number of toys where name of cat is 'Vector'
`[count(toys where name = 'Vector']   `-> **Doesn't work**

Total number of toys where the cat's friend name is 'Poo'
`[count(toys where friends.name = 'Poo') ]` -> **Doesn't work**

**Reply by DmitrySharabin** [reply]
Let's try to solve the issue step by step.

1. We need to get all the cats named Vector. Not toys, but cats. We can get the data we need this way: `[cat where name = "Vector"]`.
2. We need to access the `toys` property of all those cats, like this: `[(cat where name = "Vector").toys]`. This expression will resolve to **a collection of collections of toys**, i.e., `[[1, 5, 7]]`
3. So, to count toys for _the first cat named Vector_ we need a very complex expression: `[count(get((cat where name = "Vector").toys, 0)))]`.

However, in cases like this one, I would follow another way: I would add a computed property whose value is the needed expression and then reference this property in other expressions. For example:

```html
<ul mv-list="cat">
  <li mv-list-item>
    <meta property="toys_count" content="[count(toys)]" />

    <div property="name">Vector</div>
    <div property="age">10</div>
  </li>
</ul>

<p>
  Count of Vector's toys: [condense(cat where name = "Vector").toys_count]
</p>
```

Similarly, we can get other data.

**Reply by GalinhaLX** [reply]
Hi @DmitrySharabin and thanks for your answer

unfortunately 
[count(get((cat where name = "Vector").toys, 0)))]  **doesn't work** 

I made a codepen [here](https://codepen.io/galinhalx/pen/yLvNwEd?editors=1000) 

As I need to also edit the values of the toys, I tried with this nested structure:
(this wasn't in your answer, but I guess thats maybe what you wanted to reply?)
```
<ul mv-list="cat">
		<li mv-list-item>
			<meta property="toys_count" content="[count(toys)]" />

			<div property="name">Vector</div>
			<div property="age">10</div>
			<ul mv-list="toys">
				<li mv-list-item>

				</li>
			</ul>
		</li>
	</ul>
```

or
```
<ul mv-list="cat">
		<li mv-list-item>
			<meta property="toys_count" content="[count(toys)]" />

			<div property="name">Vector</div>
			<div property="age">10</div>
			<ul mv-list>
				<li mv-list-item="toys">

				</li>
			</ul>
		</li>
	</ul>
```

as you cans see above, I tried with 'toys' in both **mv-list** or **mv-list-item**, but I'm still not able to access it

This is just a sample, I don't need to count the number of toys :)
I need to make a request with a condition on sub nested structure

**Comment by GalinhaLX** [comment]
by the way, I realize that my first request, is on a more complex deep nested structure, and unfortunately I can not do it in a simple flat structure

in my case, i have a lot of cats, and they have a lot of friends :)

```
{
  "cat": [
    {
      "name": "Vector",
      "age": "10",
      "toys": [
        1,
        2,
        3
      ]
    },
    {
      "name": "Foo",
      "age": "99",
      "toys": [
        1,
        2,
        3
      ],
      "friends": [
        {
          "name": "Poo",
          "color": "blue"
        },
        {
          "name": "bee",
          "color": "yellow"
        }
      ]
    }
  ]
}
```

I want a condition on a level deeper than toys

**Comment by GalinhaLX** [comment, marked as answer]
Finally found it :)


[get(friends.name, 0) where cat.name ="Vector"] 

-> Shows result "Poo" as expected
